src/provider/model/AppDirectory.ts

```diff
@@ -20,6 +21,32 @@ export class AppDirectory {
         });
     }
 
+    public async getAppIntentsByContext(contextType: string): Promise<AppIntent[]> {
+        await this.fetchData();
+        const appIntentsByName: { [intentName: string]: AppIntent } = {};
+        this._directory.forEach((app: Application) => {
+            (app.intents || []).forEach(intent => {
+                if (intent.contexts && intent.contexts.includes(contextType)) {
+                    if (appIntentsByName[intent.name]) {
+                        appIntentsByName[intent.name].apps.push(app);
+                    } else {
+                        appIntentsByName[intent.name] = {
+                            intent: {
+                                name: intent.name,
+                                displayName: intent.displayName || intent.name
+                            },
+                            apps: [app]
+                        };
+                    }
+                }
+            });
+        });
+        Object.values(appIntentsByName).forEach(appIntent => {
+            appIntent.apps.sort((a, b) => a.appId.localeCompare(b.appId));
+        });
+        return Object.values(appIntentsByName).sort((a, b) => a.intent.name.localeCompare(b.intent.name));
```

Comment:
What I wanted to do here was make sure we always return the intents and the apps withing each intent in a predictable and deterministic order.  It helps with tests and I think it's nicer on the consumers of the API.  The spec doesn't say anything about order, that's true.

And you're right, at whatever presentation layer this ends up being used, that's where any smarter sorting should be done.

As for `localeCompare`, I think that's just the standard way to compare strings in JS, even if you don't care about the locales.

Reply:
Ah, sure, I agree that it should be in a predictable and deterministic order, but the result of localeCompare differs depending on the locale of the host machine, right? (And I get its unlikely to make any difference in practice, but it still feels like it would be better not to lean on a method that by design differs depending on the user's locale).

Reply:
You're right, it does depend on the language set in the browser.  As it's not a sort that would affect presentation, I've fixed by always setting the locale to `'en'` in `localeCompare`, what do you think?

Reply:
👍